prusti-specs/src/rewriter.rs

```diff
@@ -116,70 +96,105 @@ impl AstRewriter {
             &format!("prusti_{}_item_{}_{}", spec_type, item.sig().ident, spec_id),
             item_span,
         );
-        let mut statements = TokenStream::new();
-        assertion.encode_type_check(&mut statements);
         let spec_id_str = spec_id.to_string();
-        let assertion_json = crate::specifications::json::to_json_string(&assertion);
 
-        let mut spec_item: syn::ItemFn = parse_quote_spanned! {item_span=>
+        let mut spec_item: syn::ItemFn = parse_quote_spanned! {item_span =>
             #[allow(unused_must_use, unused_variables, dead_code)]
             #[prusti::spec_only]
             #[prusti::spec_id = #spec_id_str]
-            #[prusti::assertion = #assertion_json]
-            fn #item_name() {
-                #statements
+            fn #item_name() -> bool {
+                #expr
             }
         };
         spec_item.sig.generics = item.sig().generics.clone();
         spec_item.sig.inputs = item.sig().inputs.clone();
-        if spec_type == SpecItemType::Postcondition {
+        if spec_type != SpecItemType::Precondition {
```

Comment:
Does this actually make sense for a predicate?

Reply:
No, it doesn't. Fixed in 681ccfe0b2aa35aa9ed35a05b55490ba69ad34b9